compose/mpp/demo/src/uikitMain/kotlin/androidx/compose/mpp/demo/StatusBarState.kt

```diff
@@ -0,0 +1,124 @@
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.DropdownMenu
+import androidx.compose.material.DropdownMenuItem
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.derivedStateOf
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.interop.LocalUIViewController
+import androidx.compose.ui.unit.dp
+import platform.UIKit.UIStatusBarAnimation
+import platform.UIKit.UIStatusBarStyleDarkContent
+import platform.UIKit.UIStatusBarStyleDefault
+import platform.UIKit.UIStatusBarStyleLightContent
+import platform.UIKit.UIView
+
+private var statusBarStyleIndex by mutableStateOf(0)
+val preferredStatusBarStyleValue by derivedStateOf { styleValues[statusBarStyleIndex].second }
+
+private var statusBarHiddenIndex by mutableStateOf(0)
+val prefersStatusBarHiddenValue by derivedStateOf { hiddenValues[statusBarHiddenIndex].second }
+
+private var statusBarAnimationIndex by mutableStateOf(0)
+val preferredStatysBarAnimationValue by derivedStateOf { animationValues[statusBarAnimationIndex].second }
+
+private val hiddenValues = listOf(
+    "null" to null,
+    "True" to true,
+    "False" to false
+)
+
+private val animationValues = listOf(
+    "null" to null,
+    "UIStatusBarAnimationFade" to UIStatusBarAnimation.UIStatusBarAnimationFade,
+    "UIStatusBarAnimationSlide" to UIStatusBarAnimation.UIStatusBarAnimationSlide
+)
+
+private val styleValues = listOf(
+    "null" to null,
+    "UIStatusBarStyleDefault" to UIStatusBarStyleDefault,
+    "UIStatusBarStyleLightContent" to UIStatusBarStyleLightContent,
+    "UIStatusBarStyleDarkContent" to UIStatusBarStyleDarkContent
+)
+
+val StatusBarStateExample = Screen.Example("StatusBarState") {
+    Column(modifier = Modifier.fillMaxSize()) {
+        Dropdown("preferredStatusBarStyle", styleValues[statusBarStyleIndex].first, styleValues.map { it.first }) {
+            statusBarStyleIndex = it
+        }
+
+        Dropdown("prefersStatusBarHidden", hiddenValues[statusBarHiddenIndex].first, hiddenValues.map { it.first }) {
+            statusBarHiddenIndex = it
+        }
+
+        Dropdown("preferredStatysBarAnimation", animationValues[statusBarAnimationIndex].first, animationValues.map { it.first }) {
+            statusBarAnimationIndex = it
+        }
+    }
+
+    val viewController = LocalUIViewController.current
+    LaunchedEffect(statusBarStyleIndex, statusBarHiddenIndex, statusBarAnimationIndex) {
+        UIView.animateWithDuration(0.3) {
+            viewController.setNeedsStatusBarAppearanceUpdate()
```

Comment:
Just wondering if we can rework code to get rid of this call when changing status bar appearance settigns.

Reply:
We could, but I guess people know exactly when they want to do it, and it's a trivial piece of code, so better not lock ourselves into some specific behavior.